cloud-storage.go

```diff
@@ -40,39 +40,12 @@ func NewCloudStorage(
 
 	gcpCredentialsJSON string,
 	gcpStorageEmulatorHost string,
-) (CloudStorage, error) {
-	return NewCloudStorageV2(
-		ctx,
-		isTesting,
-		bucketProvider,
-		bucketName,
-
-		awsS3Endpoint,
-		awsS3Region,
-		awsS3AccessKeyID,
-		awsS3SecretAccessKey,
-		false,
-		gcpCredentialsJSON,
-		gcpStorageEmulatorHost,
-	)
-}
 
-//nolint:funlen
-func NewCloudStorageV2(
-	ctx context.Context,
-	isTesting bool,
-	bucketProvider string,
-	bucketName string,
+	opts ...*CloudStorageOption,
```

Comment:
I'm thinking for the cloud storage, do we really need multiple options? Maybe introduce an enhanced new with an option will be better. And put the existing into the option as well, such as:
		awsS3Endpoint,
		awsS3Region,
		awsS3AccessKeyID,
		awsS3SecretAccessKey,
		gcpCredentialsJSON,
		gcpStorageEmulatorHost,
		...

Reply:
I see, such as `NewCloudStorageWithOpts(opts ...*CloudStorageOption)`?
Where CloudStorageOption contains: 
awsS3Endpoint,
awsS3Region,
awsS3AccessKeyID,
awsS3SecretAccessKey,
gcpCredentialsJSON,
gcpStorageEmulatorHost
...